sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -21,6 +21,13 @@
     HttpLoggingPolicy,
     UserAgentPolicy,
 )
+from azure.core.exceptions import (
+    ClientAuthenticationError,
+    HttpResponseError,
+    ResourceNotFoundError,
+    ResourceExistsError,
+    map_error
```

Comment:
`map_error` was intended as a means of inversion-of-control (e.g. a way to inject behavior into generated code). What does using it buy us here?

Reply:
Getting specific errors like ClientAuthenticationError etc instead of generic HttpResponseError